sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/SparkSQLSessionManager.scala

```diff
@@ -50,6 +51,7 @@ private[hive] class SparkSQLSessionManager(hiveContext: HiveContext)
   }
 
   override def closeSession(sessionHandle: SessionHandle) {
+    SparkSQLEnv.sqlEventListener.onDisconnected(super.getSession(sessionHandle))
```

Comment:
I know that overriding `openSession` needs to move `SparkSQLSessionManager` into Hive shim, so it's not done here. But we should document it and explain when and where `onConnected` is called.